zookeeper-server/src/main/java/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -280,6 +306,8 @@ public void processRequest(Request request) {
                 rsp = new CreateResponse(rc.path);
                 err = Code.get(rc.err);
                 requestPathMetricsCollector.registerRequest(request.type, rc.path);
+                String createMode = getCreateMode(request);
```

Comment:
This `getCreateMode()` is quite expensive. It should be moved inside the if which checks whether audit logging is enabled. You might want to just pass the request object to `addAuditLog()` and extract in there if necessary.

Additionally we might not need AuditConstant for every single operation, but use OpCode directly. In which case audit logging could be called at one place instead of every case branch.

Reply:
Refactored complete code and this is addressed in code refactoring.

Reply:
Great, thanks!